src/main/java/Solution.java

```diff
@@ -0,0 +1,134 @@
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.StringTokenizer;
+
+@WebServlet(
+        name = "123",
+        urlPatterns = ("/calc")
+)
+public class Solution extends HttpServlet {
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws  IOException {
+        PrintWriter out = resp.getWriter();
+        String equation = req.getParameter("equation");
+        StringBuilder equationWithoutVars = new StringBuilder();
+        for (int i = 0; i < equation.length(); ++i) {
+            char cur = equation.charAt(i);
+            if (cur >= 'a' && cur <= 'z') {
+                String s = req.getParameter(String.valueOf(cur));
+                while (s.charAt(0) >= 'a' && s.charAt(0) <= 'z') {
+                    s = req.getParameter(s);
+                }
+                equationWithoutVars.append(s);
+            } else {
+                equationWithoutVars.append(cur);
+            }
+        }
+        out.print(get(equationWithoutVars.toString()));
+        out.flush();
+        out.close();
+    }
+
+    private boolean isOperator(char c) {
+        return c == '-' || c == '+' || c == '*' || c == '/';
+    }
+
+    private int priorityOfOperator(char c) {
+        switch (c) {
+            case '*':
+            case '/':
+                return 1;
+            default:
+                return 0;
+        }
+    }
+
+    private int get(String in) {
+        String equation = RPN(in);
+        int left;
+        int right;
+        String s;
+        Deque<Integer> st = new ArrayDeque<>();
+        StringTokenizer tokenizer = new StringTokenizer(equation);
+        while (tokenizer.hasMoreTokens()) {
+            s = tokenizer.nextToken().trim();
+            if (1 == s.length() && isOperator(s.charAt(0))) {
+                right = st.pop();
+                left = st.pop();
+                switch (s.charAt(0)) {
+                    case '+':
+                        left += right;
+                        break;
+                    case '-':
+                        left -= right;
+                        break;
+                    case '/':
+                        left /= right;
+                        break;
+                    case '*':
+                        left *= right;
+                        break;
+                    default:
+                        break;
+                }
+                st.push(left);
+            } else {
+                left = Integer.parseInt(s);
+                st.push(left);
+            }
+        }
+        return st.pop();
+    }
+
+
+    private String RPN(String s) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The instance method name 'RPN' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/app/thejerome/IFMO_Servlets_20190915/pullRequest?prid=4670753)